### PR TITLE
Switch to PythonCall in tests

### PR DESCRIPTION
**Pull request by mileslucas**
This removes the data generation and rewrites the tests that depend on the data using PythonCall. It is annoying to have to check-in data files and generate new data in our testing framework (e.g., #46).

Now the tests are written such that astropy coordinates are created, converted, and compared within the Julia test framework (`test/astropoy.jl`). This should be simple enough to expand for new types, and more directly compares the converted coordinates instead of the maximum allowable separation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,3 +29,4 @@ docs/site/
 Manifest.toml
 
 # End of https://www.gitignore.io/api/julia
+.CondaPkg
```
Comment (giordano) [inline .gitignore:32]: Missing newline 
```suggestion
.CondaPkg

```

File: test/runtests.jl [hunk 2]
```diff
@@ -237,3 +182,14 @@ end
...
+@testset "conversion" begin
+    systems = (ICRSCoords, FK5Coords{2000}, FK5Coords{1975}, GalCoords)
+    for IN_SYS in systems, OUT_SYS in systems
+        coord_in = IN_SYS(rand(rng), rand(rng))
+        coord_out = convert(OUT_SYS, coord_in)
+        # Test pipe and constructor conversion
+        @test coord_out == OUT_SYS(coord_in)
+        @test coord_out == coord_in |> OUT_SYS
+    end
+end
```
Comment (giordano) [inline test/runtests.jl:195]: Missing newline 
```suggestion
end

```

File: CondaPkg.toml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[deps]
+astropy = ""
```
Comment (giordano) [inline CondaPkg.toml:2]: Should specify a compatibility version number?

**Review by giordano — COMMENTED** [review]
Removing hardcoded data is nice, on the other hand this adds a test-dependency on a python package, right? I'm thinking of PkgEval, I hope this isn't a problem

**Comment by mileslucas** [comment]
> I'm thinking of PkgEval, I hope this isn't a problem

There are other JuliaAstro packages with PythonCall in tests, I think. Let's see if they're failing on PkgEval

**Comment by mileslucas** [comment]
Well, I was thinking of LombScargle, but it seems like the astropy tests aren't actually called from `runtests.jl`, so it isn't ran in CI.

**Comment by giordano** [comment]
I use pythoncall in LombScargle.jl, but that's not part of runtests.jl (so PkgEval doesn't care), it's a separate job just to check compatibility with astropy independently of the rest of the test suite, the tests here directly need astropy to generate the reference data for the unit tests.

I'm happy with the change if this isn't a problem for PkgEval

**Comment by mileslucas** [comment]
Took me a second, but here's an example of CondaPkg/PythonCall working on PkgEval

https://github.com/pabloferz/DLPack.jl

https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_date/2022-12/27/DLPack.primary.log

**Pull request merged**